### PR TITLE
Add ndarray recipe

### PR DESCRIPTION
**Pull request by brianv0**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vend other packages
- [X] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/ndarray/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - cmake
+    - m4
+    - {{ compiler('cxx') }}
+    - boost-cpp
+    - fftw
+    - numpy
+    - pybind11<2.3
+    - eigen
+    - pkgconfig
+    - python
```
Comment (scopatz) [inline recipes/ndarray/meta.yaml]: Most of these are host requirements, not build one.

File: recipes/ndarray/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+requirements:
+  build:
+    - cmake
+    - m4
+    - {{ compiler('cxx') }}
+    - boost-cpp
+    - fftw
+    - numpy
+    - pybind11<2.3
```
Comment (scopatz) [inline recipes/ndarray/meta.yaml]: This needs a single space `pybind11 <2.3`

**Comment by TallJimbo** [comment]
I (as upstream ndarray maintainer) am willing to be listed in the PR as a recipe maintainer.

**Comment by brianv0** [comment]
@conda-forge/help-c-cpp I think this is ready

**Pull request merged**